implementations/rust/ockam/ockam_api/src/authenticator/direct.rs

```diff
@@ -171,10 +201,20 @@ where
                         if tkn.time.elapsed() > MAX_TOKEN_DURATION {
                             api::forbidden(&req, "expired token").to_vec()?
                         } else {
-                            let attributes = minicbor::to_vec(&tkn.attrs)?;
-                            self.store
-                                .set(from.key_id(), MEMBER.to_string(), attributes)
-                                .await?;
+                            //TODO: fixme:  unify use of hashmap vs btreemap
+                            let attrs = tkn
+                                .attrs
+                                .iter()
+                                .map(|(k, v)| (k.to_string(), v.as_bytes().to_vec()))
+                                .collect();
+                            let entry = AttributesEntry::new(
```

Comment:
I think this deserves a constructor for `AttributesEntry` taking a map and an `IdentityIdentifier` to remove the duplication.

Reply:
agree,  will let it as TODO to address latter.    _Ideally_  we should settle on use either  btreemap or hashmap  for this and avoid the conversions.   And it's mostly about ergonomics,   as the cardinality of this map is not going to be large.    (I _think_  why btree map was introduced on some parts was because it serialized out-of-the-box through cbor,  but I might be wrong on that)